### ISSUE TITLE
Deserialization converters on JsonCreator parameters

### ISSUE THREAD

**Issue by elucash** [FasterXML/jackson-databind#821]
I have a situation when I'm not sure if it should work this way or not: so when I use `@JsonCreator` method, deserialization converter on parameter does not work as it works on the a field.

Here's an example where map have custom serializer and deserialization converter. A map is being marshaled as array of name value pairs in JSON. However, on deserialization, converter which consume an array of entries and produces the map is not being picked up. Map deserializer tries to deal with START_ARRAY and fails.

```java
public class DataValue {
  private final Map<String, String> attributes;
  private DataValue(Map<String, String> attributes) {
    this.attributes = attributes;
  }
  @JsonCreator(mode = JsonCreator.Mode.PROPERTIES)
  public static DataValue from(
      @JsonDeserialize(converter = JsonSetToMapConverter.class)
      @JsonProperty("a") Map<String, String> attributes) {
    return new DataValue(attributes);
  }
  @JsonSerialize(using = MapJsonSerializer.class)
  @JsonProperty("a")
  public Map<String, String> getAttributes() {
    return attributes;
  }
}
```

The following simple example with annotated field which works as expected:

```java
public class DataValue {
  @JsonDeserialize(converter = JsonSetToMapConverter.class)
  @JsonSerialize(using = MapJsonSerializer.class)
  @JsonProperty("a")
  public Map<String, String> attributes;
}
```

Should I expect `JsonDeserialize converter` to work with `JsonCreator`? Is it bug?
Thank you

For some of the details you can view executable example files
https://www.dropbox.com/s/sc8tk31gjkjak0l/example.zip?dl=1
`MainTest` is the entry point there

**Comment by cowtowncoder** [comment]
Yes, those should work similarly, unless I am missing something.

Thank you for reporting this, I will try to see why is the converter being ignored.

One thing to try is to see if `2.6.0-rc2` (which was just released) has the same bug, assuming you may be using version 2.5.x?

**Comment by elucash** [comment]
Thank you for suggestion, it works properly after upgrading to `2.6.0-rc2` from `2.5.x`.
I'm closing this issue.

**Comment by cowtowncoder** [comment]
Thank you for verifying this. I did fix one problem that was related to problem keeping 2 copies of creator-backed properties in sync, so I am assuming this is the root cause.
I will mark this as fixed in 2.6.0-rc2 then.
